client/state/sites/domains/selectors.js

```diff
@@ -41,3 +46,31 @@ export const isRequestingSiteDomains = ( state, siteId ) => {
 	const { requesting } = state.sites.domains;
 	return requesting[ siteId ] || false;
 };
+
+/**
+ * Returns decorated site domains with objects we don't want to store in Redux state tree.
+ *
+ * @param  {Object}         state global state
+ * @param  {Number}  siteId the site id
+ * @return {Object|null}    decorated site domains
```

Comment:
We can describe nullable types like so: `@return {?Object}` see: http://usejsdoc.org/tags-type.html